ignite/handlers/timing.py

```diff
@@ -0,0 +1,141 @@
+import time
+from ignite.engine import Events
+
+
+class Timer:
+    """ Timer object can be used to measure (average) time between events.
+
+    Args:
+        unit (bool, optional): if True, then when ``.value()`` method is called, the returned value
+            will be equal to total time measured, divided by the value of internal counter.
+
+    Attributes:
+        total (float): total time elapsed when the Timer was running (in seconds)
+        cnt (int): internal counter, usefull to measure average time, e.g. of processing a single batch.
+            Incremented with the ``.inc_unit()`` method.
+        running (bool): flag indicating if time is measuring time.
+
+    Notes:
+        When using ``Timer(unit=True)`` do not forget to call ``timer.inc_unit()`` everytime an event occurs. See
+        the examples below.
+
+    Examples:
+        Measuring total time of the entire 'training' epoch:
+
+        >>> from ignite.handlers import Timer
+        >>> import time
+        >>> work = lambda : time.sleep(0.1)
+        >>> idle = lambda : time.sleep(0.1)
+        >>> t = Timer(unit=False)
+        >>> for _ in range(10):
+        ...    work()
+        ...    idle()
+        ...
+        >>> t.value()
+        2.003073937026784
+
+        Measuring average time of the 'training' epoch:
+        >>> t = Timer(unit=True)
+        >>> for _ in range(10):
+        ...    work()
+        ...    idle()
+        ...    t.inc_unit()
+        ...
+        >>> t.value()
+        0.2003182829997968
+
+        Measuring average time it takes to execute a single ``work()`` call
+        >>> t = Timer(unit=True)
+        >>> for _ in range(10):
+        ...    t.resume()
+        ...    work()
+        ...    t.pause()
+        ...    idle()
+        ...    t.inc_unit()
+        ...
+        >>> t.value()
+        0.10016545779653825
+
+        Using the Timer to measure average time it takes to process a single batch of examples
+        >>> from ignite.trainer import Trainer
+        >>> from ignite.engine import Events
+        >>> from ignite.handlers import Timer
+        >>> trainer = Trainer(trainig_update_function)
+        >>> timer = Timer(unit=True)
+        >>> timer.attach(trainer,
+        ...              start=Events.EPOCH_STARTED,
+        ...              resume=Events.ITERATION_STARTED
+        ...              pause=Events.ITERATION_COMPLETED,
+        ...              step=Events.ITERATION_COMPLETED)
+    """
+
+    def __init__(self, unit=False):
```

Comment:
should we rename `unit` to `mean` or `average`, I know `unit` is used in tnt but I still can't understand why